### ISSUE TITLE
undefined function render_block/2 when combining mode: :live_view and slot in component

### ISSUE THREAD

**Issue by rktjmp** [mhanberg/temple#133]
**Describe the bug**

Setting `config :temple, :mode, :live_view` breaks slotted component.

**To Reproduce**

- Clone https://github.com/rktjmp/temple61-liveview-slot-bug
- `mix phx.server`
- observe error:
```
== Compilation error in file lib/bug_web/live/user_live/a_component.ex ==
** (CompileError) lib/bug_web/live/user_live/a_component.ex:29: undefined function render_block/2
    (elixir 1.11.4) src/elixir_locals.erl:114: anonymous fn/3 in :elixir_locals.ensure_no_undefined_local/3
    (stdlib 3.14.1) erl_eval.erl:680: :erl_eval.do_apply/6
    (elixir 1.11.4) lib/kernel/parallel_compiler.ex:314: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/7
```

- edit `lib/bug_web/live/user_live/a_component.ex`, comment out line 33
- `mix phx.server`, observe no error
- edit `lib/bug_web/live/user_live/a_component.ex`, uncomment line 33
- edit `config/config.ex`, comment out line 29 (or set mode to `:normal`)
- `mix phx.server`, observe no error

`lib/bug_web/live/user_test/test.*` and  `localhost:4000/test` use the component.

I think I have it configured correctly, I have a working app that stops working when I flip the `:mode` to `:live_view`. 

The mix file might be a bit of a mess as I threw whatever it asked to resolve the dependency tree.

**Desktop**
 - OS: Arch Linux
 - Temple Version 6.1
 - Elixir Version 1.11.4
 - Erlang Version  23.3.1

**Comment by mhanberg** [comment]
Sorry you ran into this, I need to improve the documentation around this.

you need to make sure that the `Phoenix.LiveView.Helpers` are imported into your component modules.
What i've done in my app is add a `component` "clause" into `lib/my_app_web.ex`

```elixir
  def component do
    quote do
      import Temple.Component

      unquote(view_helpers())
    end
  end
```

and then i write my component like

```elixir
defmodule MyAppWeb.Component.SomeComponent do
  use MyAppWeb, :component

  ...
end
```

**Comment by rktjmp** [comment]
Yep that fixes it.

1)

On the topic, having a `normal` vs `live_view` config setting sort of implies they are exclusive, but normal views *seem* to work fine in live view mode.

I can understand that live views in normal mode might not get the diff tracking but is there a downside to just setting it to `mode: :live_view`, assuming you have at least one live view in the app? The code defaults to normal, probably for a reason.

Maybe that should be in the docs too?

2) 

The requirement to track phoenix_live_view HEAD could maybe be better explained by perhaps linking to a specific SHA that has the changes needed so I can judge when those changes have made it into hex instead of you having to track that and refresh the docs.

I will leave you to close this issue if you want to use it to track the docs task.

**Comment by mhanberg** [comment]
Closed by #177
